sumo-api/sumo/metrics/utils.py

```diff
@@ -242,6 +242,47 @@ def compute_pr(det_matches, det_scores, n_gt, recall_samples=None, interp=False)
 
     return (precision, recall)
 
+def compute_ap(det_matches, det_scores, n_gt):
```

Comment:
I suggest instead of writing a competing function compute_ap, that you integrate this into the compute_pr function that has a flag to determine if you want to sample the PR curve at fixed locations or compute the PR at the data points.  If you don't like that suggestion, you could create an alternative compute_pr function: e.g., compute_sampled_pr -- the old one -- and compute_exact_pr -- the new one.  

The benefit of doing this is that the other existing code will still work (e.g., plot_pr) if you keep the interface the same.  Also, the code is more general if you keep the function of computing the PR separate from the AP computation.

Reply:
compute_ap is the latest implementation of PASCAL VOC2010+ that uses AUC. When we use this, compute_pr and related functions like plot_pr etc. become obsolete as we will not use them, so both should be removed. There are no other parts of the code/evaluation where they will be needed once the update is made and I don't see any reason why we would want to go back to an older version of the PASCAL metrics in the future. compute_ap is also cleaner and less complex than compute_pr's implementation and I don't think it makes sense to add complexity to support something that is already obsolete. It makes more sense to just delete compute_pr and plot_pr.

Reply:
I don't think the other code is obsolete, particularly because it is quite useful for people to be able to visualize the PR curves.  Also, we may decide that the other method for computing the PR curve is better.  I have seen other implementations of the same concept that keep both methods.  It does not really add any complexity to the code, but it gives us the option of computing both of the most common approaches.

Reply:
I believe we already decided over email (when I sent the references) to update to the most recent Pascal implementation as we discussed that it was a generalization of compute_pr where the number of sampled points is infinite. And if people really want to visualize PR curves, we can just return the precision and exact recall values that are in compute_ap.